clang/lib/Driver/ToolChains/NanoMips.cpp

```diff
@@ -124,7 +124,7 @@ void NanoMipsLinker::ConstructJob(Compilation &C, const JobAction &JA,
     ToolChain.addFastMathRuntimeIfAvailable(Args, CmdArgs);
   }
 
-  if (Args.hasArg(options::OPT_mrelax)) {
+  if (Args.hasArg(options::OPT_mrelax, options::OPT_mno_relax, true)) {
```

Comment:
So after this change, we would still be able to disable relaxation with `--no-relax option`?

Reply:
This matches gcc behaviour [1] where absence of -mno-relax means pass --relax to the linker otherwise nothing.

[1] https://github.com/MediaTek-Labs/gcc/blob/mips/umips/gcc6_v12/gcc/config/mips/nanomips.h#L214

Reply:
Correct, the `hasArg` argumetns are "positive option", "negative option" and "default". It resolves the last of either the positive or negative values  and returns true or false depending on which was last, or the default if neither.

Without this change, we're currently doing the wrong thing. If we pass `-mrelax` ... `-mnorelax` the LLVM back-end will correctly get the `-relax`  (ie. disable it) flag but the linker will get `--relax`.

Reply:
OK, makes sense.